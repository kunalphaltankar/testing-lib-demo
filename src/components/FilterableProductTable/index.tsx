import { useState } from "react";
import { IProduct } from "../../models";
import ProductTable from "../ProductTable";
import SearchBar from "../SearchBar";

function FilterableProductTable({ products }: { products: IProduct[] }) {
  const [state, setState] = useState({
    filterText: '',
    inStockOnly: false,
  });

  function handleFilterTextChange(filterText: string) {
    setState({
      ...state,
      filterText: filterText,
    });
  }

  function handleInStockChange(inStockOnly: boolean) {
    setState({
      ...state,
      inStockOnly: inStockOnly,
    });
  }

  return (
    <div>
      <SearchBar
        filterText={state.filterText}
        inStockOnly={state.inStockOnly}
        onFilterTextChange={handleFilterTextChange}
        onInStockChange={handleInStockChange}
      />
      <ProductTable
        products={products}
        filterText={state.filterText}
        inStockOnly={state.inStockOnly}
      />
    </div>
  );
}

export default FilterableProductTable;