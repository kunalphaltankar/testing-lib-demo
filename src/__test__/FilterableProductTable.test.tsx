import { render } from "@testing-library/react";
import FilterableProductTable from "../components/FilterableProductTable";
import PRODUCTS from "../models/stub/PRODUCTS";

describe("FilterableProductTable", () => {
  it('should render FilterableProductTable', () => {
    const { container } = render(<FilterableProductTable products={PRODUCTS} />);
    expect(container).toMatchSnapshot();
  })
})