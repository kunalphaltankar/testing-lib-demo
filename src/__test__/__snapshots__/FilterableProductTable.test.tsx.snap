// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FilterableProductTable should render FilterableProductTable 1`] = `
<div>
  <div>
    <form>
      <input
        placeholder="Search..."
        type="text"
        value=""
      />
      <p>
        <input
          type="checkbox"
        />
         
        Only show products in stock
      </p>
    </form>
    <table>
      <thead>
        <tr>
          <th>
            Name
          </th>
          <th>
            Price
          </th>
        </tr>
      </thead>
      <tbody>
        <tr>
          <th
            colspan="2"
          >
            Sporting Goods
          </th>
        </tr>
        <tr>
          <td>
            Football
          </td>
          <td>
            $49.99
          </td>
        </tr>
        <tr>
          <td>
            Baseball
          </td>
          <td>
            $9.99
          </td>
        </tr>
        <tr>
          <td>
            <span
              style="color: red;"
            >
              Basketball
            </span>
          </td>
          <td>
            $29.99
          </td>
        </tr>
        <tr>
          <th
            colspan="2"
          >
            Electronics
          </th>
        </tr>
        <tr>
          <td>
            iPod Touch
          </td>
          <td>
            $99.99
          </td>
        </tr>
        <tr>
          <td>
            <span
              style="color: red;"
            >
              iPhone 5
            </span>
          </td>
          <td>
            $399.99
          </td>
        </tr>
        <tr>
          <td>
            Nexus 7
          </td>
          <td>
            $199.99
          </td>
        </tr>
      </tbody>
    </table>
  </div>
</div>
`;
